readabilipy/extractors/extract_element.py

```diff
@@ -0,0 +1,35 @@
+from bs4 import BeautifulSoup
+from ..text_manipulation import normalise_whitespace
+from ..plain_html import unwrap_elements
+import lxml.html
+
+
+def extract_element(html, xpaths):
+    """Return the relevant element (title, date or byline) from article HTML
+        xpaths should be a list of tuples, each with the xpath and a reliability score
+    """
+
+    # Discard certain html elements while keeping contents
+    soup = BeautifulSoup(html)
+    unwrap_elements(soup)
+    html = soup.prettify()
+
+    # Parse the html and extract the contents using all xpaths
+    lxml_html = lxml.html.fromstring(html)
+    elements = []
```

Comment:
I think this could be a bit simpler. We can make a dictionary of `extracted text => score` which we fill as we go along. We can then take the maximum of this.